src/offset/local/unix.rs

```diff
@@ -24,19 +24,25 @@ pub(super) fn offset_from_local_datetime(local: &NaiveDateTime) -> MappedLocalTi
 
 fn offset(d: &NaiveDateTime, local: bool) -> MappedLocalTime<FixedOffset> {
     TZ_INFO.with(|maybe_cache| {
-        maybe_cache.borrow_mut().get_or_insert_with(Cache::default).offset(*d, local)
+        maybe_cache.borrow_mut().offset(*d, local)
     })
 }
 
-// we have to store the `Cache` in an option as it can't
-// be initialized in a static context.
 thread_local! {
-    static TZ_INFO: RefCell<Option<Cache>> = Default::default();
+    static TZ_INFO: RefCell<Cache> = const { RefCell::new(
+        Cache {
```

Comment:
Why? This doesn't seem like an improvement.

Reply:
It is a preparation for the next commit: splitting up the `offset` function.
The old code did all in the initialization in `offset()` using `Cache::default` on first use within a thread.

The `offset_from_utc_datetime` and `offset_from_local_datetime` both had to go through the `offset` function even though they had little in common (and preferably not the same return type), so they could use the same `get_or_insert_with`.

The goal of this commit is to move the concern of initialization from the caller the `Cache` type.